### ISSUE TITLE
enhance libpsm2 version check to allow building against older libpsm2 under certain circumstances

### ISSUE THREAD

**Issue by dshrader** [open-mpi/ompi#8381]
Thank you for taking the time to submit an issue!

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

v4.1.0

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

4.1.0 release tarball

### Please describe the system on which you are running

* Operating system/version: based on RHEL 7
* Computer hardware: Intel
* Network type: omnipath

-----------------------------

## Details of the problem

When building on a RHEL 7 based system, the psm2 and ofi mtls can't be built as the version of libpsm2.so is below version 11.2.173. The latest version of libpsm2 available for RHEL 7 and CentOS 7 is 11.2.78, which means the psm2 and ofi mtls can't be built on stock RHEL 7 or CentOS 7 without patching ompi's configure.

In looking through the commit that supplies the libpsm2 version check and talking with @hppritcha, it seems the real concerning error condition is `with-ofi=yes` AND `with-psm2=yes` AND `libpsm2 version < 11.2.173`. If either ofi or psm2 is turned off, then there shouldn't be an issue. Would it be possible to enhance the libpsm2 version check such that it won't error out if only one of the ofi or psm2 mtls is being built so that these components can still be used on stock RHEL and CentOS 7?

Maybe something like the following could work?
```
if libpsm2 version >= 11.2.173
    nothing needed
else
    if with-ofi=yes AND with-psm2=yes
        error to user; they have to choose one or update libpsm2
    if with-ofi=yes AND with-psm2=unset
        with-psm2=no
    if with-ofi=unset AND with-psm2=yes
        with-ofi=no
    if with-ofi=unset AND with-psm2=unset
        error or automatically choose a default?
```

**Comment by rhc54** [comment]
@mwheinz Any thoughts?

**Comment by mwheinz** [comment]
Which release of RHEL 7? We definitely continue to support the current version of PSM2 for RHEL 7.8 and 7.9. 

Have you considered grabbing the latest release of PSM2 from https://github.com/cornelisnetworks/opa-psm2 and using that?

**Comment by dshrader** [comment]
We're running RHEL 7.8 on our HPC clusters, but we don't pull rpms directly from RedHat update servers. On a RedHat 7.9 machine that is pulling from RedHat update servers, the latest libpsm2 that I can pull is 11.2.78-1.

For this scenario, I am actually building Open MPI 4.1.0 for the general user base of our HPC clusters. We have considered providing hardware communication libraries such as libpsm2 ourselves, and while I have done that for debugging purposes, we have decided that the MPI builds destined for the general user base should rely on the vendor-supplied hardware communication libraries. So, I can get it to work for myself, but not for the builds that our general users are going to pound on.

**Comment by mwheinz** [comment]
> For this scenario, I am actually building Open MPI 4.1.0 for the general user base of our HPC clusters.

In that case I should warn you that there's a known bug with 4.1.0 and PSM2 that will prevent you from working at all. The fix is going to be in 4.1.1.

**Comment by dshrader** [comment]
Which bug? For my own testing, I patched the version check out of 4.1.0's configure and one of the psm2 mtl files, and built with only the psm2 mtl enabled, and I was able to run the Intel MPI Benchmark suite as well as I normally could. I'm curious about what workload I may be missing in my testing...

**Comment by mwheinz** [comment]
#8305.

Another way to get a newer PSM would be to get the latest IFS package from (checks notes) [Intel's website](https://www.intel.com/content/www/us/en/design/products-and-solutions/networking-and-io/fabric-products/omni-path/product-releases-library.html). (We're still in the process of transitioning back to being an independent company so things are in flux right now...)

**Comment by mwheinz** [comment]
All that said, I'm not entirely opposed to your request - if you want to submit a patch we can take a look at it.

**Comment by rhc54** [comment]
@mwheinz Just FYI: @dshrader is using the PSM MTL, not the OFI one, and so #8305 does not apply.

**Comment by hppritcha** [comment]
closed via #8552 and #8590

**Comment by hppritcha** [comment]
turns out we want this in the 4.1.x release branch too.

**Comment by hppritcha** [comment]
closed via #8921
